stage1/init/init.go

```diff
@@ -605,36 +605,65 @@ func stage1() int {
 		log.FatalE("error making / a shared and slave mount", err)
 	}
 
-	enabledCgroups, err := cgroup.GetEnabledCgroups()
+	unifiedCgroup, err := cgroup.IsCgroupUnified()
 	if err != nil {
-		log.FatalE("error getting cgroups", err)
+		log.FatalE("error determing cgroup version", err)
 		return 1
 	}
 
-	if err := mountHostCgroups(enabledCgroups); err != nil {
-		log.FatalE("couldn't mount the host cgroups", err)
-		return 1
-	}
-
-	var serviceNames []string
-	for _, app := range p.Manifest.Apps {
-		serviceNames = append(serviceNames, stage1initcommon.ServiceUnitName(app.Name))
-	}
 	s1Root := common.Stage1RootfsPath(p.Root)
 	machineID := stage1initcommon.GetMachineID(p)
-	subcgroup, err := getContainerSubCgroup(machineID)
-	if err == nil {
-		if err := ioutil.WriteFile(filepath.Join(p.Root, "subcgroup"),
-			[]byte(fmt.Sprintf("%s", subcgroup)), 0644); err != nil {
-			log.FatalE("cannot write subcgroup file", err)
+	if unifiedCgroup {
```

Comment:
If `unifiedCgroup` is true, I don't think we have to do anything:
- writing the "subcgroup" was only necessary to cleanup the cgroups during `rkt gc` in the non-systemd hierarchies (e.g. /sys/fs/cgroup/*memory*/...) but we don't have those hierarchies in the unified case.
- we don't need to do the `mountContainerCgroups` and we can let systemd-nspawn do that.